csharp/src/Ice/OpaqueEndpoint.cs

```
@@ -111,6 +105,13 @@ protected internal override void AppendOptions(StringBuilder sb, char optionSepa
             }
         }
 
+        protected internal override Endpoint Clone(string host) =>
```

Can we just throw `InvalidOperationException`, seems this is only used to compute the OA published endpoints and it cannot contain opaque endpoints.

---

I think we should keep the implementation. In the future we may find another use case.

---

Added a TODO.

---

Removed TODO and replaced Clone by GetPublishedEndpoint that throws NotSupportedException as suggested by Jose/Joe.